channels/models.py

```diff
@@ -1,15 +1,81 @@
-"""Models for channels_fields"""
+"""Models for channels"""
 
 from django.contrib.auth.models import Group
+from django.core.validators import RegexValidator
 from django.db import models
-from django.db.models import deletion
+from django.db.models import JSONField, deletion
+from imagekit.models import ImageSpecField, ProcessedImageField
+from imagekit.processors import ResizeToFit
 
-from channels_fields.constants import FIELD_ROLE_CHOICES
-from discussions.models import BaseChannel
+from channels.constants import FIELD_ROLE_CHOICES
 from learning_resources.models import LearningResource
 from open_discussions.models import TimestampedModel
+from profiles.utils import avatar_uri, banner_uri
 from widgets.models import WidgetList
 
+AVATAR_SMALL_MAX_DIMENSION = 22
+AVATAR_MEDIUM_MAX_DIMENSION = 90
+
+
+class BaseChannel(models.Model):
+    """Base abstract model for channels"""
```

Comment:
This is fine with me as an abstract model, but I wonder if @ferdi might prefer that we refactor this similarly to what we did with `LearningResource` (which also used to be abstract).  So we would end up with one channels API with potentially different `channel_type` values (like `field_channel`) that could be filtered on.  That is a pretty major change though, and if the API for this app isn't meant to be public-facing like the learning_resources API , then I feel like it wouldn't be worth it.